tempodb/encoding/vparquet3/block_traceql.go

```diff
@@ -1314,7 +1321,12 @@ func createSpanIterator(makeIter makeIterFn, primaryIter parquetquery.Iterator,
 
 	// Left join here means the span id/start/end iterators + 1 are required,
 	// and all other conditions are optional. Whatever matches is returned.
-	return parquetquery.NewLeftJoinIterator(DefinitionLevelResourceSpansILSSpan, required, iters, spanCol)
+	j, err := parquetquery.NewLeftJoinIterator(DefinitionLevelResourceSpansILSSpan, required, iters, spanCol)
+	if err != nil {
+		return nil, err
+	}
+	j.Pool = pqSpanPool.Get
```

Comment:
Not a huge fan of this, feels a bit unsafe. But IDK if another approach is worth it.

Reply:
👍 Switched to options pattern.